lms/lmsdb/models.py

```diff
@@ -447,13 +479,18 @@ def view_solution(self) -> None:
     def start_checking(self) -> bool:
         return self.set_state(Solution.STATES.IN_CHECKING)
 
-    def set_state(self, new_state: SolutionState, **kwargs) -> bool:
+    def set_state(
+        self, new_state: SolutionState,
+        evaluation: Optional[SolutionEvaluation] = None, **kwargs,
+    ) -> bool:
         # Optional: filter the old state of the object
         # to make sure that no two processes set the state together
         requested_solution = (Solution.id == self.id)
+        updates_dict = {Solution.state.name: new_state.name}
+        if evaluation is not None:
+            updates_dict[Solution.evaluation.name] = evaluation
         changes = Solution.update(
-            **{Solution.state.name: new_state.name},
-            **kwargs,
+            **updates_dict, **kwargs,
         ).where(requested_solution)
```

Comment:
Join to a single line?

Reply:
Can't because it's going to be 86 characters in this single line